src/lcmtype.jl

```diff
@@ -203,3 +330,108 @@ encode(x::LCMType) = (stream = IOBuffer(false, true); encode(stream, x); flush(s
 
 decode!(x::LCMType, data::Vector{UInt8}) = decode!(x, BufferedInputStream(data))
 decode(data::Vector{UInt8}, ::Type{T}) where {T<:LCMType} = decode!(T(), data)
+
+# @lcmtypesetup macro and related functions
+function make_fixed_dimensions_method(::Type{T}, ::Type{<:SVector{N, <:Any}}, field::Symbol, axisnum::Int) where {T<:LCMType, N}
+    let lcmdim = LCMCore.LCMDimension(N)
+        LCMCore.dimensions(::Type{T}, ::Val{field}, ::Val{axisnum}) = lcmdim
+    end
+end
+
+function make_fixed_dimensions_methods(::Type{T}, ::Type{F}, field::Symbol, axisnum::Int) where {T<:LCMType, V, F<:AbstractVector{V}}
+    F <: SVector && make_fixed_dimensions_method(T, F, field, axisnum)
+    V <: AbstractVector && return make_fixed_dimensions_methods(T, V, field, axisnum + 1)
+    axisnum
+end
+
+function make_dimensions_methods(::Type{T}) where T<:LCMType
+    for field in fieldnames(T)
+        F = fieldtype(T, field)
+        numaxes = F <: AbstractVector ? make_fixed_dimensions_methods(T, F, field, 1) : 0
+        fieldval = Val(field)
+        let dimtuple = ntuple(i -> Base.invokelatest(LCMCore.dimensions, T, fieldval, Val(i)), numaxes)
```

Comment:
It took me a few minutes to figure out what was going on here and why `invokelatest` is required. I assume it's because multiple methods are being defined within the same function call and they need to access methods from the previous iterations of the loop. Is that right? In any case, could you add a comment?

Reply:
Yeah, that's exactly it. I'll add a comment.